src/test/java/com/suse/saltstack/netapi/client/SaltStackMockConnection.java

```diff
@@ -0,0 +1,35 @@
+package com.suse.saltstack.netapi.client;
+
+import com.google.gson.Gson;
+import com.suse.saltstack.netapi.exception.SaltStackException;
+
+import java.lang.reflect.Type;
+
+/**
+ * Class that implements a mocked connection. Can be configured to return a valid JSON or an Exception.
+ */
+class SaltStackMockConnection implements SaltStackConnection {
+    private final String mockJson;
+    private final SaltStackException mockException;
+
+    public SaltStackMockConnection(String mockJson) {
+        this.mockJson = mockJson;
+        this.mockException = null;
+    }
+
+    public SaltStackMockConnection(SaltStackException mockException) {
+        this.mockJson = null;
+        this.mockException = mockException;
+    }
+
```

Comment:
I am puzzled here why we would need these _two_ constructors _either_ way? What would be wrong with only one constructor (in case we would need the entire class at all, which I am not sure at the moment)?
```java
public SaltStackMockConnection(String mockJson, SaltStackException mockException) {
   ...
}

SaltStackMockConnection a = new SaltStackMockConnection(null, ....);
SaltStackMockConnection b = new SaltStackMockConnection(..., null);